### PR TITLE
feature(simulator): support an external scheduler

### LINKED ISSUE #182: Support external scheduler
There is great value for this project to support many types of schedulers.

Just replacing [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/9de8c472f348b31437cce5ca2a34506f874cdddb/scheduler/scheduler.go#L86) to create user's scheduler is the easy way for users to use their scheduler in simulator, but, if user's scheduler is based on different version of Kubernetes or built on a completely different mechanism than kube-scheduler in k/k, it doesn't work.

For those schedulers, I propose here to support "external scheduler". 

Users can run any type of scheduler as outside schedulers and that scheduler will communicate with kube-apiserver in the simulator to run simulation on the simulator.

There is drawback to outside scheduler: they cannot change scheduler configuration via Web UI (or simulator API).
But, the big advantage of this is users can use their scheduler with the other advanced features of the simulator like Scenario.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/140

/kind feature
/triage accepted
/priority important-longterm

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/area web
/area simulator
/kind feature

#### What this PR does / why we need it:

support external schedulers.
When EXTERNAL_SCHEDULER_ENABLED is given as true, the simulator doesn't start a scheduler and web UI also doesn't show the gear button like this, because we cannot get and change an external scheduler's configuration.

スクリーンショット 2022-09-11 17 37 23

I'll create the doc for users to use an external scheduler later in another PR.

#### Which issue(s) this PR fixes:

part of #182

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: simulator/config/config.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ type Config struct {
 	// This field is non-empty only when ExternalImportEnabled == true.
 	ExternalKubeClientCfg *rest.Config
 	InitialSchedulerCfg   *v1beta2config.KubeSchedulerConfiguration
+	// ExternalSchedulerEnabled indicates whether an external scheduler is enabled.
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply simulator/config/config.go:34]: prefer mine.

File: web/store/schedulerconfiguration.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import { reactive, inject } from "@nuxtjs/composition-api";
-import { SchedulerConfiguration } from "~/api/v1/types";
+import {KubeSchedulerProfile, SchedulerConfiguration} from "~/api/v1/types";
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply web/store/schedulerconfiguration.ts]: 👍 
KubeSchedulerProfile seems not used, so just removed it.

**Comment by sanposhiho** [comment]
> We also need to consider the Reset feature.

🙇 
Fixed

**Comment by 196Ikuchil** [comment]
Thanks!
/lgtm

**Pull request merged**